bebraven/platform
# Add SurveyVersion < CustomContentVersion

### Description

**Task**: https://app.asana.com/0/1174274412967132/1198925268705965

This PR builds on #402, which added Surveys. In order to submit Impact Surveys, we need a `SurveyVersion`.

This PR adds the `SurveyVersion`, which uses single-table inheritance from `CustomContentVersion` (like `ProjectVersion`). It also updates the pundit policies so both `Project` and `SurveyVersionPolicy` delegate to `CustomContentVersion` policy. 

Survey version uses single-table inheritance, storing its data with project versions in the table `custom_content_version`:
<img width="1474" alt="Screen Shot 2020-10-29 at 1 49 19 PM" src="https://user-images.githubusercontent.com/62911934/97631466-437b5880-19ee-11eb-89cb-3bc3c03e9eb4.png">


### Motivation and Context

When a Fellow submit an Impact Survey, we need that submission to be tied to a particular `SurveyVersion`. This is so that we can continue to render the responses in the form even if we later update the contents of the `Survey`. 

This is analogous to what we do for `Project`, `ProjectVersion` and `ProjectSubmission`, described [here](https://github.com/bebraven/platform/wiki/Project-Submissions).

### Test

- Added/updated specs
- Create a new Survey in the Content Editor and **Save**:
<img width="1792" alt="Screen Shot 2020-10-28 at 3 03 56 PM" src="https://user-images.githubusercontent.com/62911934/97502116-b7a2f700-192f-11eb-8a3f-37bbc8728b39.png">

- Replace `/edit` with **`/versions`** in the URL and see that there aren't any versions yet:
<img width="1792" alt="Screen Shot 2020-10-28 at 3 04 01 PM" src="https://user-images.githubusercontent.com/62911934/97502137-c1c4f580-192f-11eb-9b11-1de262913be2.png">

- Using the Rails console, create a new version of that survey:

```
me = User.find(9) # any admin-user will work
survey = Survey.last 
survey.save_version!(me)
```

- Refresh the page and see that the new version appears:
<img width="1792" alt="Screen Shot 2020-10-28 at 3 05 46 PM" src="https://user-images.githubusercontent.com/62911934/97502189-e1f4b480-192f-11eb-96ce-c591e5fbffaa.png">

- Click the version number and see the survey version:

<img width="1792" alt="Screen Shot 2020-10-28 at 3 05 50 PM" src="https://user-images.githubusercontent.com/62911934/97502201-e5883b80-192f-11eb-8bbb-5077a15a9b3c.png">

### Types of changes




### Checklist:
